addons/stock/models/product.py

```diff
@@ -191,6 +194,27 @@ def _compute_quantities_dict(self, lot_id, owner_id, package_id, from_date=False
 
         return res
 
+    def _compute_nbr_moves(self):
+        res = defaultdict(dict)
+        incoming_moves = self.env['stock.move.line'].read_group([('product_id', 'in', self.ids),
+                                                                 ('state', '=', 'done'),
+                                                                 ('picking_code', '=', 'incoming'),
+                                                                 ('date', '>=', fields.Datetime.now() - relativedelta(years=1))],
+                                                                ['product_id'], ['product_id'])
```

Comment:
Please don't do that :crying_cat_face: 
I know we can ignore the PEP8 rule about line length, but it doesn't worth to do that to avoid one or two linebreak, and it doesn't help te readability on Github or when you divide your workspace into multiple pads.
I would prefer:
```suggestion
        incoming_moves = self.env['stock.move.line'].read_group([
            ('product_id', 'in', self.ids),
            ('state', '=', 'done'),
            ('picking_code', '=', 'incoming'),
            ('date', '>=', fields.Datetime.now() - relativedelta(years=1))],
            ['product_id'], ['product_id'])
```
... or even:
```suggestion
        incoming_moves = self.env['stock.move.line'].read_group(
            [
                ('product_id', 'in', self.ids),
                ('state', '=', 'done'),
                ('picking_code', '=', 'incoming'),
                ('date', '>=', fields.Datetime.now() - relativedelta(years=1))
            ], ['product_id'], ['product_id']
        )
```

Reply:
Anyway, I think you can use a `search_count` here too:
```python3
        for product in self:
            domain = [
                ('product_id', '=', product.id),
                ('state', '=', 'done'),
                ('date', '>=', fields.Datetime.now() - relativedelta(years=1)),
            ]
            product.nbr_moves_in = self.env['stock.move.line'].search_count(
                expression.AND([domain, [('picking_code', '=', 'incoming')]])
            )
            product.nbr_moves_out = self.env['stock.move.line'].search_count(
                expression.AND([domain, [('picking_code', '=', 'outgoing')]])
            )
```

Reply:
I'm not sure anymore... You had only two `read_group` for all products with your solution, and two `search_count` by product with mine.
Currently it doesn't seem you can compute on multiple product in one time, but that could change and I think your version will be better in this case.

Reply:
Oh no, I don't want to make cats cry! I fixed the formatting.